### PR TITLE
Item Frames, Full implementation

### PR DESCRIPTION
**Pull request by wow4201**
Complete implementation of item frames

-Placement against blocks.
-1.8 (8 degrees of item rotation in frame).
-Item frame and item inside item frame drop when the block it's attached to is destroyed.
-Place item in frame.
-Saving/Loading.
-Can remove item inside of item frame, can destroy item frame.
-Removes item from inventory when placing inside of item frame.

This is a fully tested and complete implementation of item frames.

### CODE REVIEW

File: src/main/java/net/glowstone/entity/objects/GlowItemFrame.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+            player.setItemInHand(is);
+
+            metadata.set(MetadataIndex.ITEM_FRAME_ROTATION, 0);
+            metadata.set(MetadataIndex.ITEM_FRAME_ITEM, new ItemStack(Material.AIR));
+            metadata.set(MetadataIndex.AIR_TIME, 300);
+            iteminframe = Material.AIR;
+        }
+    }
+
+    public void SetItemInFrame(ItemStack is) {
```
Comment (ZephireNZ) [inline src/main/java/net/glowstone/entity/objects/GlowItemFrame.java]: Shouldn't this be 'setItemInFrame'?

File: src/main/java/net/glowstone/entity/objects/GlowItemFrame.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+            player.setItemInHand(is);
+
+            metadata.set(MetadataIndex.ITEM_FRAME_ROTATION, 0);
+            metadata.set(MetadataIndex.ITEM_FRAME_ITEM, new ItemStack(Material.AIR));
+            metadata.set(MetadataIndex.AIR_TIME, 300);
+            iteminframe = Material.AIR;
+        }
+    }
+
+    public void SetIteminFrame(ItemStack is) {
```
Comment (TomCaserta) [inline src/main/java/net/glowstone/entity/objects/GlowItemFrame.java:52]: He meant change the capital "S" in Set to "set" so the complete method name would be: "setItemInFrame"

Comment (ZephireNZ) [inline src/main/java/net/glowstone/entity/objects/GlowItemFrame.java:52]: Yep, it's almost universal java naming standards.

**Comment by wow4201** [comment]
Stopped developing for glowstone until something of mine gets pulled

**Comment by Postremus** [comment]
@turt2live  said that he couldn't merge your other pull requests. YOU should go to him and talk to him.

**Comment by daboross** [comment]
@wow4201 As far as I can see, all of the pull requests are waiting on changes from you before being merged.

It might be a good idea to read some of the comments on the pull requests and make the changes, or comment to ask for more information about them.

I think you've fixed some of the things in some of them, but there are still unresolved things.

**Comment by dequis** [comment]
Also, there are 27 open pull requests and limited review time. Just in case this isn't obvious enough, 27 pull requests is a massive amount of code to review. 50528 lines of patches, specifically. And only turt2live and spacemaniac are doing in-depth reviews and testing of that. Complaining about stuff not getting merged is childish in a project like this one.

File: src/main/java/net/glowstone/entity/objects/GlowItemFrame.java [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+
+public final class GlowItemFrame extends GlowEntity implements ItemFrame {
+
+    BlockFace face;
+    Material iteminframe;
+    int rot = 0;
+
```
Comment (Wild1145) [inline src/main/java/net/glowstone/entity/objects/GlowItemFrame.java:30]: Extra line here and line 143

**Comment by wow4201** [comment]
None of this is being "fixed" until something is pulled so I know my time isn't wasted here. Sorry, It may be childish but I am volunteering my time here.

I just don't see a lot of hope for this project if management can't keep up with contributions. The rate of contributions I feel is low for that matter, so I don't see it getting any better.

I have no experience managing a Github project, nor do I get any say in the way this project is managed.
I'm just laying out my expectations for management. If they are not met, that's completely fine, I will just find myself useful elsewhere. I've sort of got that vibe anyways so perhaps maybe just show my self out the door.

I'm just saying this project got a lot of interest, and people are just leaving because they are encountering the same feeling I have here.

**Comment by dequis** [comment]
50k lines of patches to review is low. Got it.

**Comment by wow4201** [comment]
Yes well it's not a surprise things pile up when they are not attended to.

**Comment by wow4201** [comment]
I personally feel this "In depth" analysis of pull requests is wasting everyone's time, when things don't get pulled because of an extra line, or tin

**Comment by dequis** [comment]
Welcome to github. It is your resposibility as a contributor to address the issues in your code. Nothing unusual was asked from you. This is perfectly normal here. In fact, turt2live and SpaceManiac are working much harder than the average maintainer you'd expect here.

Things didn't "pile up", this project got an absurd amount of contributions in the last two weeks and all of them need to be handled appropriately. You wouldn't want any crap to get merged and have a glowstone that throws NPEs every 5 seconds, would you?

Anyway, a while ago I was going over your pull requests to complete the stuff you refused to do. In one of them, I just had to run a simple rebase command to get a readable diff of a trivial looking patch, as opposed to a 3k line nonsense. Stuff like that.

However, your attitude makes me reconsider why I even waste my time trying to help you get your contributions merged. Whatever.

**Pull request closed without merging**